### PR TITLE
Fix: Include permission management on permanently delete, rename, and restore.

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
Expands what was done in https://github.com/WordPress/gutenberg/pull/62589 to other three actions where it makes sense: Permanently delete, rename, and restore post action.

## Testing Instructions
I moved a page to the trash.
I added the following code to a PHP file:
```
function remove_delete_post_capability( $caps, $cap, $user_id, $args ) {
    if ( 'delete_post' == $cap ) {
        // Get the current user object
        $current_user = wp_get_current_user();
        // Check if the current user is the user we want to remove the capability from
        if ( $current_user->ID == $user_id ) {
            // Remove the 'delete_post' capability
            // It's done by returning a capability that the user does not have, for example 'do_not_allow'
            $caps[] = 'do_not_allow';
        }
    }

    return $caps;
}

add_filter( 'map_meta_cap', 'remove_delete_post_capability', 10, 4 );
```
I went to the trash page views and verified the permanently delete page action is unavailable.
I went to drafts view and verified the trash post action is still not available.

I added the following code to a PHP file:
```
function remove_edit_post_capability( $caps, $cap, $user_id, $args ) {
    if ( 'edit_post' === $cap ) {
        // Get the current user object
        $current_user = wp_get_current_user();
        // Check if the current user is the user we want to remove the capability from
        if ( $current_user->ID == $user_id ) {
            // Remove the 'delete_post' capability
            // It's done by returning a capability that the user does not have, for example 'do_not_allow'
            $caps[] = 'do_not_allow';
        }
    }

    return $caps;
}

add_filter( 'map_meta_cap', 'remove_edit_post_capability', 10, 4 );
```

I went to the drafts pages list and verified the ability to rename a page was not shown.
I went to the trash pages list and verified the ability to restore pages was not present.

### CODE REVIEW

File: packages/editor/src/components/post-actions/actions.js [hunk 1]
```diff
@@ -535,6 +551,14 @@ const restorePostAction = {
 	},
 };
 
+function useRestorePostAction( postType ) {
+	return useCanUserEligibilityCheckPostType(
+		postType,
+		'update',
+		restorePostAction
```
Comment (ellatrix) [inline packages/editor/src/components/post-actions/actions.js:545]: Well, turns out minimising the diff was useful for reusability too 😉

File: packages/editor/src/components/post-actions/actions.js [hunk 2]
```diff
@@ -307,7 +307,7 @@ const trashPostAction = {
 	},
 };
 
-function useTrashPostAction( postType ) {
+function useCanUserEligibilityCheckPostType( postType, capability, action ) {
 	const registry = useRegistry();
 	const { resource, cachedCanUserResolvers } = useSelect(
 		( select ) => {
```
Comment (ellatrix) [inline packages/editor/src/components/post-actions/actions.js]: Could we avoid creating this subscription many times, and just do it once?

**Comment by ellatrix** [comment]
When I remove the edit cap, I'm still seeing the Edit button:

image

**Comment by jorgefilipecosta** [comment]
> When I remove the edit cap, I'm still seeing the Edit button:
> 
> image

Hi @ellatrix,

I think that's ok, the edit means open in the site editor, but in my tests the site editor does permission checks and for example, if you open a page there but don't have permission to edit you will not be able to change the page content. I guess if you have permission to change the template you may change the template but not the content.

File: packages/editor/src/components/post-actions/actions.js (hunk 1 above)
Comment (jorgefilipecosta) [reply packages/editor/src/components/post-actions/actions.js:545]: Yes for sure the diff change made things easier to reuse.

File: packages/editor/src/components/post-actions/actions.js (hunk 2 above)
Comment (jorgefilipecosta) [reply packages/editor/src/components/post-actions/actions.js]: Hi @ellatrix, I applied a change and now this PR does not add any store subscription.

**Review by ellatrix — APPROVED** [review]

File: packages/editor/src/components/post-actions/actions.js [hunk 3]
```diff
@@ -307,34 +307,29 @@ const trashPostAction = {
...
-			...trashPostAction,
+			...action,
 			isEligible( item ) {
 				return (
-					trashPostAction.isEligible( item ) &&
+					action.isEligible( item ) &&
 					registry
 						.select( coreStore )
-						.canUser( 'delete', resource, item.id )
+						.canUser( capability, resource, item.id )
```
Comment (ellatrix) [inline packages/editor/src/components/post-actions/actions.js:320]: I guess it doesn't really happen in the case, but the problem here is that `isEligible` will not be called if the user capability changes, right? So maybe all of these need to be precalculated/subscribed to at some point so the buttons re-render.

**Comment by ellatrix** [comment]
Handled in #62823.

**Pull request merged**